pkg/kubeapiserver/authorizer/reload.go

```diff
@@ -140,3 +159,120 @@ func (r *reloadableAuthorizerResolver) newForConfig(authzConfig *authzconfig.Aut
 
 	return union.New(authorizers...), union.NewRuleResolvers(ruleResolvers...), nil
 }
+
+// runReload starts checking the config file for changes and reloads the authorizer when it changes.
+// Blocks until ctx is complete.
+func (r *reloadableAuthorizerResolver) runReload(ctx context.Context) {
+	metrics.RegisterMetrics()
+	metrics.RecordAuthorizationConfigAutomaticReloadSuccess(r.apiServerID)
+
+	go func() {
+		if err := r.watchFile(ctx); err != nil {
+			select {
+			case <-ctx.Done():
+				// server was shutting down, we don't care about reload errors
+			default:
+				klog.ErrorS(err, "watching authorization config file")
```

Comment:
Should this restart the watch?

Reply:
not sure... I don't really know the variety of issues that could cause errors on the file watch, and don't want to do anything problematic with loops or exhausting file handles, so if it hits errors, I'd tend to collapse to the poll

Reply:
updated to make the poll attempt to restart the watch if an error was observed